### PR TITLE
Tidy up requirements, all in setup.py now

### PR DESCRIPTION
**Pull request by znicholls**
# Pull request

Please confirm that this pull request has done the following:

- [x] Tests added (N/A)
- [x] Documentation added (where applicable) (N/A)
- [x] Example added (either to an existing notebook or as a new notebook, where applicable) (N/A)
- [x] Description in ``CHANGELOG.rst`` added

## Adding to CHANGELOG.rst

Please add a single line in the changelog notes similar to one of the following:

```
- (`#XX <http://link-to-pr.com>`_) Added feature which does something
- (`#XX <http://link-to-pr.com>`_) Fixed bug identified in (`#XX <http://link-to-issue.com>`_)
```

### CODE REVIEW

**Comment by rgieseke** [comment]
Good job! Can we pin expectexcept to a specific commit?

**Comment by rgieseke** [comment]
Another question, `expectexceptation` is not really needed for running, it should go into tests, right?
`Appmode` and `ipywidgets` is only needed for running the Mybinder demo. We could also move it to a separate project (scm-demos?) ...

File: setup.py [hunk 1]
```diff
@@ -48,19 +52,26 @@ def run_tests(self):
...
+    "expectexception @ git+https://github.com/thedataincubator/expectexception.git@master",
+     "ipywidgets",
+     "appmode",
 ]
 
-extra_requirements = {"test": ["pytest", "pytest-cov", "codecov", "goodtables"]}
+extra_requirements = {
+    "tests": ["pytest>=4.0", "pytest-cov", "codecov", "goodtables", "notebook", "matplotlib", "nbval", "goodtables"],
+    "docs": ["sphinx>=1.4", "sphinx_rtd_theme", "sphinx-autodoc-typehints", "pydoc-markdown"],
+    "deploy": ["setuptools>=38.6.0", "twine>=1.11.0", "wheel>=0.31.0", "flake8", "black", "versioneer"],
```
Comment (rgieseke) [inline setup.py]: Maybe we should have a separate group `dev` or call the whole thing `dev` as only the first three are needed for PyPI.

Versioneer is not needed here i think, it's more a system tool that creates the necessary files in your project. Or is it needed to update `versioneer` from time to time?

**Comment by znicholls** [comment]
> Can we pin expectexcept to a specific commit?

Yes. Might be a touch slow (https://stackoverflow.com/questions/13685920/install-specific-git-commit-with-pip) but fine until one of us gets around making a request for a release in their repo.

**Comment by znicholls** [comment]
> Another question, `expectexceptation` is not really needed for running, it should go into tests, right?
> `Appmode` and `ipywidgets` is only needed for running the Mybinder demo. We could also move it to a separate project (scm-demos?)

I thought about this and my feeling is that most people look at the notebooks when they use pymagicc. Hence all of the stuff someone needs to run the notebooks should be put as a dependency. Having people run notebooks and then get import errors just seems to be asking for trouble so I'd prefer to leave expectexception etc. in package dependencies...

File: setup.py (hunk 1 above)
Comment (znicholls) [reply setup.py]: > Maybe we should have a separate group `dev` or call the whole thing `dev` as only the first three are needed for PyPI.

Ok.

> Or is it needed to update `versioneer` from time to time?

Nope I don't think so, have removed.

**Comment by rgieseke** [comment]
> Yes. Might be a touch slow (https://stackoverflow.com/questions/13685920/install-specific-git-commit-with-pip) but fine until one of us gets around making a request for a release in their repo.

Interesting. For this small repo it shouldn't be a big problem - i also realize now that it's not such a big difference to other libs on PyPI as we also don't pin them.

It still feels clearer somehow :-)

**Comment by znicholls** [comment]
> It still feels clearer somehow :-)

Haha whatever works

**Pull request merged**